chatlearn/utils/vllm_utils.py

```diff
@@ -821,8 +826,16 @@ def convert_llama_state_dict_from_megatron_to_vllm(args, hf_config, qwen_version
 
     # Transformer Layers
     print("Converting transformer layers")
-    num_layers = hf_config.num_hidden_layers // pp_size
-
+    if CURRENT_VLLM_VERSION == VLLMVersion.v_0_6_3:
+        start_layer_idx, _ = get_pp_indices(
```

Comment:
we needed build pp_indices from  the mapping by training to inference，and then set envs.VLLM_PP_LAYER_PARTITION for vllm.

Reply:
Done. We need to set envs.VLLM_PP_LAYER_PARTITION before model setup, With config num_layers and pipeline_model_parallel_size, we infer num of layers for each stage when allow_padding_num_layers set True.